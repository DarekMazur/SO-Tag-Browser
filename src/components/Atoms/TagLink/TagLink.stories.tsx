import type { Meta, StoryObj } from '@storybook/react';
import { faker } from '@faker-js/faker';
import TagLink from './TagLink.tsx';

const meta: Meta<typeof TagLink> = {
  tags: ['autodocs'],
  component: TagLink,
  title: 'Atoms/TagLink',
};

export default meta;

type Story = StoryObj<typeof TagLink>;

export const Default: Story = {
  argTypes: {
    to: {
      control: 'text',
      description: 'StackOverflow tags endpoint',
      name: 'tag',
    },
  },
  args: {
    to: faker.lorem.word(),
  },
};
