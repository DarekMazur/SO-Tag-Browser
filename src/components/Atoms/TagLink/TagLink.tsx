import { FC } from 'react';
import LaunchIcon from '@mui/icons-material/Launch';
import { Link } from '@mui/material';
import { orange } from '@mui/material/colors';
import { useTranslation } from 'react-i18next';
import { ITagLink } from '../../../types/types.ts';

const TagLink: FC<ITagLink> = ({ to }) => {
  const { t } = useTranslation();

  return (
    <Link
      href={`https://stackoverflow.com/questions/tagged/${to}`}
      target="_blank"
      rel="noreferrer"
      sx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'flex-end',
        gap: '0.5rem',
        textDecoration: 'none',
        fontWeight: 500,
        color: orange[900],
        '&:hover': {
          color: orange[700],
        },
      }}
    >
      {t('table.link', { tag: to })} <LaunchIcon fontSize="small" />
    </Link>
  );
};

export default TagLink;
