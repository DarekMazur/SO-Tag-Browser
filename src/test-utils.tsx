// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { ReactNode } from 'react';
import AppProviders from './providers/AppProviders.tsx';

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return <AppProviders>{children}</AppProviders>;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const customRender = (ui, options?) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';

export { customRender as render };
