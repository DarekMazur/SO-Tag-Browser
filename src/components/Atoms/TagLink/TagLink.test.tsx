import { render, screen } from '../../../test-utils';
import TagLink from './TagLink.tsx';

describe('Tag link', () => {
  it('Renders the component', () => {
    const { container } = render(<TagLink to="example" />);
    expect(container).toMatchSnapshot();
  });
  it('Renders with prop', () => {
    render(<TagLink to="example" />);
    expect(screen.getByText(/example/)).toBeInTheDocument();
  });
});
