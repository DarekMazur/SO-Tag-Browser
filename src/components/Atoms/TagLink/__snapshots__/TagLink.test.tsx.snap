// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag link > Renders the component 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1nkbz67-MuiTypography-root-MuiLink-root"
    href="https://stackoverflow.com/questions/tagged/example"
    rel="noreferrer"
    target="_blank"
  >
    see posts for example
     
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-y3v3gx-MuiSvgIcon-root"
      data-testid="LaunchIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
      />
    </svg>
  </a>
</div>
`;
